src/lib/THERMAL/devThermal.cpp

```diff
@@ -86,7 +117,29 @@ static void timeoutFan()
     {
         if (fanShouldBeOn)
         {
+#if defined(PLATFORM_ESP32)
+            if (GPIO_PIN_FAN_PWM != UNDEF_PIN)
+            {
+                static PowerLevels_e lastPower = MinPower;
+                if (POWERMGNT::currPower() < lastPower && fanStateDuration < FAN_MIN_CHANGETIME)
+                {
+                    ++fanStateDuration;
+                }
+                if (POWERMGNT::currPower() > lastPower || (POWERMGNT::currPower() < lastPower && fanStateDuration >= FAN_MIN_CHANGETIME))
+                {
+                    ledcWrite(fanChannel, fanSpeeds[POWERMGNT::currPower()]);
+                    DBGLN("Fan speed: %d (power) -> %d (pwm)", POWERMGNT::currPower(), fanSpeeds[POWERMGNT::currPower()]);
+                    lastPower = POWERMGNT::currPower();
+                    fanStateDuration = 0; // reset the timeout
+                }
+            }
+            else
+            {
+                fanStateDuration = 0; // reset the timeout
+            }
```

Comment:
bump this block of code to outside the ifdef since it is the same I think?

Reply:
I could if I early exited the function at the line above where it's incremented.

Reply:
I was thinkin' like this would support both ESP32 / everything else. It just makes it slightly easier to read for me, so if you're not down with that, you can just mark this as resolved. Everything I tagged was so minor 😁
```cpp
#if defined(PLATFORM_ESP32)
            if (GPIO_PIN_FAN_PWM != UNDEF_PIN)
            {
...
            }
            else
#else
            {
                fanStateDuration = 0; // reset the timeout
            }
#endif
```